### ISSUE TITLE
Some currently unsupported cases

### ISSUE THREAD

**Issue by arnaud-lb** [phpstan/phpstan-mockery#3]
### Some expectation methods

```
$mock = Mockery::mock(Something::class);
$mock
    ->shouldReceive('foo')
    ->andReturn(null)
    ->shouldReceive('bar');
```

Currently, the second `shouldReceive` call will trigger this error: `Call to an undefined method Mockery\Expectation::shouldReceive().`

### Partial mocks

```
$mock = Mockery::mock('Foo\Bar[method1,method2]');
```

Currently, $mock is assumed to be of type `Foo\Bar[method1,method2]`

### Methods returning Mock

```
$mock = Mockery::mock(Bar::class)->shouldDeferMissing();
```

Currently, methods returning Mockery\Mock will just return that, instead of an intersection type.

**Comment by jaydiablo** [comment]
Also doesn't currently support the alias mock syntax:

```php
Mockery::mock('alias:Foo\Bar')
            ->shouldReceive('bar')
```

>Call to method shouldReceive() on an unknown class alias:Foo\Bar.

**Comment by ondrejmirtes** [comment]
Solved with #6.
